libbpf-cargo/src/build.rs

```diff
@@ -242,11 +242,11 @@ fn compile(debug: bool, objs: &[UnprocessedObj], clang: &Path, target_dir: &Path
     Ok(())
 }
 
-fn extract_clang_or_default(clang: Option<&PathBuf>) -> PathBuf {
-    match clang {
+fn extract_command_or_default(command: Option<&PathBuf>, default: &str) -> PathBuf {
+    match command {
         Some(c) => c.into(),
         // Searches $PATH
-        None => "clang".into(),
+        None => default.into(),
     }
 }
```

Comment:
Please don't adjust this function like this. It's already questionable at best: the name is misleading and it basically is an `Option::unwrap_or_else` in disguise and with confusing and borderline nonsensical details. Proper way is:

```patch
--- src/build.rs
+++ src/build.rs
@@ -286,7 +286,7 @@ pub fn build_single(
     clang: Option<&PathBuf>,
     skip_clang_version_checks: bool,
     options: &str,
-    bpftool: Option<&PathBuf>,
+    bpftool: Option<&Path>,
     vmlinux_h: Option<&PathBuf>,
 ) -> Result<()> {
     let clang = extract_command_or_default(clang, "clang");
@@ -301,7 +301,7 @@ pub fn build_single(

     if let Some(vmlinux_h) = vmlinux_h {
         let file = fs::File::create(vmlinux_h)?;
-        let bpftool = extract_command_or_default(bpftool, "bpftool");
+        let bpftool = bpftool.unwrap_or_else(|| Path::new("bpftool"));
         Command::new(bpftool)
             .arg("btf")
             .arg("dump")
diff --git src/lib.rs src/lib.rs
index 73e145..dce9a5 100644
--- src/lib.rs
+++ src/lib.rs
@@ -289,7 +289,7 @@ impl SkeletonBuilder {
             self.clang.as_ref(),
             self.skip_clang_version_check,
             &self.clang_args,
-            self.bpftool.as_ref(),
+            self.bpftool.as_deref(),
             self.vmlinux_h.as_ref(),
         )
         .map_err(Error::Build)?;
```

Reply:
Thanks; applied